README.md

```diff
@@ -17,31 +21,94 @@ Works well with [redux-optimist-promise](https://github.com/mathieudutour/redux-
   npm install --save reduck
 ```
 
+
 ## Usage
 
+Working with ducks means reducers are split into types of data and each reducer (or ~duck~) defines all the actions that are relevant to a specific type of data (e.g. to-do items).
+
+Let's look at a basic example:
+
+*actions.js*
+```js
+export const FETCH_TODOS = 'todo.FETCH_TODOS';
+export const ADD_TODO = 'todo.ADD_TODO';
+export const LOGOUT = 'auth.LOGOUT';
+```
+The package expects the action name to be defined as `<duck>.<action-name>`.
+This helps with tracking defined actions in each duck.
+
+Now let's look at what the todo items duck will look like:
+
+*ducks/todos.js*
 ```js
 import Duck from 'reduck'
 
 import {
+  ADD_TODO,
   FETCH_TODOS,
   LOGOUT,
 } from '../actions';
 
+// define an initial state to use in the Duck's initialization
 const initialState = {
   items: [],
   ready: false,
 };
 
 const duck = new Duck('todo', initialState);
+```
+We start by simply creating a `new Duck`. The resulting object gives us 2 methods that we can use to define action creators and corresponding reducers: 
+`defineAction(actionName: String, reducerCases: Object)`
+- `actionName` is the name that will be given to the action. Can be whatever you choose.
+- `reducerCases` consist of:
+  * `creator(actionArgs)`
+  The creator accepts any arguments related to the action being performed and then returns the object that will be passed to the action's reducers. The object consists of mainly `payload` (the payload that will be handled by the reducers) and `meta` (data that will be used by middleware). 
+  _Note:_The creator **must** be present in the `defineAction` method.
+  * `reducer(state, { payload })`
```

Comment:
here i would show only `reducer` and in another section, explain how it fits nicely with `redux-object-to-promise`